src/components/WrappedActionMenu/ActionTemplate/ActionTemplate.tsx

```diff
@@ -202,20 +184,32 @@ export const ActionTemplate: React.FC<IProps> = ({
           alignItems='flex-end'
           wrap='nowrap'
           className={classes.secondHalf}>
-          <OutlinedButton onClick={onMaxButtonClick} className={classes.maxButton} name='Max' />
+          {
+            isXs &&
+          <OutlinedButton
+            name={capitalizeString(action)}
+            disabled={!actionAvailable}
+            color='secondary'
+            className={classNames(classes.actionButton, classes.actionButtonXS)}
+            onClick={onClick(amountBN, decimal)}
+            labelClassName={classes.label}
+          />
+          }
           <Divider orientation='vertical' className={classes.divider} />
           <Grid item className={classes.available}>
             <KeyValue
               keyName={`Available to ${action}`}
               keyClassName={classes.textCenter}
               valueClassName={classes.textCenter}
-              value={maxAvailable}
-              decimal={maxDecimal}
-              unit={currency}
+              unit={tokenToIndex ? tokens[tokenToIndex]?.symbol : currency}
```

Comment:
maybe currency, maxAvailable and maxDecimal should be also optional props or just use list with one token and add prop to input, that will control if there should be modal with list showing on click